packages/screens/RiotGame/RiotGameFightScreen.tsx

```diff
@@ -1,201 +1,135 @@
 export const RiotGameFightScreen = () => {
   const navigation = useAppNavigation();
-  const { setToastError } = useFeedbacks();
-  const { myAvailableRippers } = useRippers();
-
-  const [isShowClaimModal, setIsShowClaimModal] = useState(false);
-  const [isUnstaking, setIsUnstaking] = useState(false);
-
-  const [stakedRippers, setStakedRippers] = useState<RipperLightInfo[]>([]);
 
   const {
-    currentSquad,
+    squads,
     squadStakingConfig,
     currentUser,
     squadWithdraw,
-    remainingTime,
-    stakingState,
-    startStakingTimer,
-    lastStakeTime,
-    isLastStakeTimeLoaded,
-    isStakingStateLoaded,
-    isSquadLoaded,
-    setCurrentSquad,
+    isSquadsLoaded,
+    fetchSquads,
   } = useSquadStaking();
 
-  const fetchCurrentStakedRippers = async (currentStakedNfts: Nft[]) => {
-    const client = await getNonSigningCosmWasmClient();
-
-    const nftInfos: NftInfoResponse[] = await Promise.all(
-      currentStakedNfts.map((nft) =>
-        client.queryContractSmart(nft.contract_addr, {
-          nft_info: { token_id: nft.token_id },
-        })
-      )
-    );
+  const [now, setNow] = useState<number>(0);
 
-    const stakedRippers: RipperLightInfo[] = nftInfos.map(({ extension }) => ({
-      imageUri: ipfsURLToHTTPURL(`${extension?.image}`),
-      name: `${extension?.name}`,
-    }));
+  const isCompleted = (squad: Squad) => now - squad.end_time * 1000 >= 0;
 
-    setStakedRippers(stakedRippers);
-  };
-
-  const unstake = async () => {
-    if (!currentUser) return;
-
-    try {
-      setIsUnstaking(true);
-
-      await squadWithdraw(currentUser);
-      setIsShowClaimModal(true);
-    } catch (e: any) {
-      setToastError({
-        title: "Error occurs",
-        message: e.message,
-      });
-    } finally {
-      setIsUnstaking(false);
-    }
-  };
+  const ongoingSquads = useMemo(
+    () => squads.filter((m) => !isCompleted(m)),
+    [now]
+  );
+  const completedSquads = useMemo(() => squads.filter(isCompleted), [now]);
 
   const gotoMarketplace = () => {
     navigation.navigate("RiotGameMarketplace");
   };
 
   const onCloseClaimModal = () => {
-    setIsShowClaimModal(false);
-    setCurrentSquad(undefined);
+    fetchSquads(currentUser || "");
   };
 
-  /*
-  - If there is squad, in all case we need to keep the fight screen
-  - If there is no squad:
-    + If state is Completed/Unknown => go to enroll screen
-    + If state is Relax => show fight screen with countdown base on lastStakeTime
-  */
   useEffect(() => {
-    if (
-      isSquadLoaded &&
-      !currentSquad &&
-      isStakingStateLoaded &&
-      [StakingState.COMPLETED, StakingState.UNKNOWN].includes(stakingState)
-    ) {
-      navigation.replace("RiotGameEnroll");
-    }
-  }, [isSquadLoaded, isStakingStateLoaded]);
-
-  // Start the timer
-  useEffect(() => {
-    if (!isSquadLoaded || !squadStakingConfig?.owner || !isLastStakeTimeLoaded)
-      return;
+    const interval = setInterval(() => {
+      setNow(+new Date());
```

Comment:
`Date.now()` achieves the same thing and is less cryptic

Reply:
Yep, use Date.now() instead of +new Date()